domino/domino.py

```diff
@@ -525,15 +525,41 @@ def collaborators_get(self):
         url = self._routes.collaborators_get()
         return self._get(url)
 
-    def collaborators_add(self, usernameOrEmail, message=""):
+    def get_user_id(self, username_or_email): 
+        url = self._routes.users_get()
+        response = self.request_manager.get(url)
+        users = response.json()
+        user_from_user_name = [x for x in users if ('userName' in x) and (x['userName'] == username_or_email)]
+        user_from_email = [x for x in users if ('email' in x) and (x['email'] == username_or_email)]
+        user_list = user_from_user_name + user_from_email
+        if len(user_list) == 0:
+            raise Exception(f"Could not find user: {username_or_email}")
```

Comment:
With regard to the expected behavior when a user is not present, I don't have authoritative experience on the subject, but I do know that other similar API's that provide a similar lookup (e.g., Keycloak's API also lets you look up a user ID by user name, or even just any basic DB lookup) doesn't throw an exception if a record matching the given search term is not found. Typically, returning None or an empty set is sufficient. Throwing an Exception seems unexpected to me.